rules/retrieve.smk

```diff
@@ -3,376 +3,557 @@
-    datafiles = [
-        "IGGIELGN_LNGs.geojson",
-        "IGGIELGN_BorderPoints.geojson",
-        "IGGIELGN_Productions.geojson",
-        "IGGIELGN_Storages.geojson",
-        "IGGIELGN_PipeSegments.geojson",
-    ]
+if (COSTS_DATASET := dataset_version("costs"))["source"] in [
+    "primary",
+]:
 
-    rule retrieve_gas_infrastructure_data:
+    rule retrieve_cost_data:
+        input:
+            costs=storage(COSTS_DATASET["url"] + "/costs_{year}.csv"),
         output:
-            expand("data/gas_network/scigrid-gas/data/{files}", files=datafiles),
-        log:
-            "logs/retrieve_gas_infrastructure_data.log",
-        retries: 2
-        conda:
-            "../envs/environment.yaml"
-        script:
-            "../scripts/retrieve_gas_infrastructure_data.py"
+            costs=COSTS_DATASET["folder"] / "costs_{year}.csv",
+        run:
+            copy2(input["costs"], output["costs"])
 
 
-if config["enable"]["retrieve"]:
+if (POWERPLANTS_DATASET := dataset_version("powerplants"))["source"] in [
+    "primary",
+]:
 
-    rule retrieve_electricity_demand:
-        params:
-            versions=["2019-06-05", "2020-10-06"],
+    rule retrieve_powerplants:
+        input:
+            powerplants=storage(POWERPLANTS_DATASET["url"]),
         output:
-            "data/electricity_demand_raw.csv",
-        log:
-            "logs/retrieve_electricity_demand.log",
-        resources:
-            mem_mb=5000,
-        retries: 2
-        conda:
-            "../envs/environment.yaml"
-        script:
-            "../scripts/retrieve_electricity_demand.py"
+            powerplants=f"{POWERPLANTS_DATASET['folder']}/powerplants.csv",
+        run:
+            copy2(input["powerplants"], output["powerplants"])
+
 
+if (SCIGRID_GAS_DATASET := dataset_version("scigrid_gas"))["source"] in [
+    "primary",
+    "archive",
+]:
 
-if config["enable"]["retrieve"]:
+    rule retrieve_gas_infrastructure_data:
+        input:
+            zip_file=storage(SCIGRID_GAS_DATASET["url"]),
+        output:
+            zip_file=f"{SCIGRID_GAS_DATASET['folder']}/IGGIELGN.zip",
+            entry=f"{SCIGRID_GAS_DATASET['folder']}/data/IGGIELGN_BorderPoints.geojson",
+            storage=f"{SCIGRID_GAS_DATASET['folder']}/data/IGGIELGN_Storages.geojson",
+            gas_network=f"{SCIGRID_GAS_DATASET['folder']}/data/IGGIELGN_PipeSegments.geojson",
+        run:
+            copy2(input["zip_file"], output["zip_file"])
+            output_folder = Path(output["zip_file"]).parent
+            unpack_archive(output["zip_file"], output_folder)
+
+
+rule retrieve_electricity_demand:
+    params:
+        versions=["2019-06-05", "2020-10-06"],
+    output:
+        "data/electricity_demand_raw.csv",
+    log:
+        "logs/retrieve_electricity_demand.log",
+    resources:
+        mem_mb=5000,
+    retries: 2
+    conda:
+        "../envs/environment.yaml"
+    script:
+        "../scripts/retrieve_electricity_demand.py"
```

Comment:
Could save as `csv.gz`

Reply:
Closing as not planned - premature optimisation ;)